internal/datasource/git.go

```diff
@@ -31,20 +36,66 @@ func (s *GitSource) ProjectSource(body hcl.Body, ctx *hcl.EvalContext) (*pb.Job_
 		return nil, diag
 	}
 
+	// Start building the result
+	result := &pb.Job_Git{
+		Url:  cfg.Url,
+		Path: cfg.Path,
+	}
+	switch {
+	case cfg.Username != "":
+		result.Auth = &pb.Job_Git_Basic_{
+			Basic: &pb.Job_Git_Basic{
+				Username: cfg.Username,
+				Password: cfg.Password,
+			},
+		}
+
+	case cfg.SSHKey != "":
+		result.Auth = &pb.Job_Git_Ssh{
+			Ssh: &pb.Job_Git_SSH{
+				PrivateKeyPem: []byte(cfg.SSHKey),
+				Password:      cfg.SSHKeyPassword,
+			},
+		}
+	}
```

Comment:
Should we check here that the given url and auth mechanism works?

Reply:
We don't really have to. The validation here isn't as bad as it appears. During `terraform init` we schedule a `ValidateOp` job and that should also validate that the cloning works properly. I think the key feedback here is that we should schedule this job as part of the UI workflow and maybe add a `waypoint project validate` CLI to the project CLI proposal as well.

Reply:
Gotcha, right. Sounds good!